addons/pos_restaurant/static/src/overrides/components/ticket_screen/ticket_screen.js

```diff
@@ -126,9 +126,12 @@ patch(TicketScreen.prototype, {
     async onDoRefund() {
         const order = this.getSelectedOrder();
         if (this.pos.config.module_pos_restaurant && order && !this.pos.selectedTable) {
-            await this.pos.setTable(
-                order.table ? order.table : this.pos.models["restaurant.table"].getAll()[0]
-            );
+            const table = order.table
+                ? order.table
+                : this.pos.models["restaurant.table"].getAll()[0];
+            if (table) {
+                await this.pos.setTable(table);
+            }
```

Comment:
```suggestion
            const tableModel = this.pos.models["restaurant.table"];
            const table = order.table_id || tableModel.getFirst();
            if (table) {
                await this.pos.setTable(table);
            }
```

To access table from order object is `table_id`

Reply:
You can also maybe add a test